packages/extract-react-types/src/utils.js

```diff
@@ -0,0 +1,25 @@
+/* eslint-disable import/prefer-default-export */
+
+export function hasTypeAnnotation(path, left, right) {
```

Comment:
I really like this function ❤️. Got lost in `if` checks though 😅

Reply:
Yeah it's a bit of a hairy one, isn't it 🤔 
Can you see any obvious ways to simplify it? Maybe a change of approach might be best here instead.

Reply:
Yeah! Don't have full context but can we do something like this (not sure if this is correct, just a nit TBH 😅):
We can also skip `try` `catch` I think 🤔

```js
const { typeName, typeParameters } = path.get('id.typeAnnotation.typeAnnotation').node;

return !!typeParameters && [
  typeName.left?.name === left && typeName.right?.name === right,
  typeName.name === left || typeName.name === right,
].some(Boolean);
```

Reply:
Updated 😄 

I managed to remove the `try/catch` and simplify the logic a bit. 

Hopefully it's looking better. LMK what you think